### PR TITLE
Use new  msgpack package

### PR DESCRIPTION
**Pull request by gernest**
This PR introduces github.com/ugorji/go/codec  as mspack encoder/decoder

### CODE REVIEW

File: ably/realtime_client_test.go [hunk 1]
```diff
@@ -90,7 +90,7 @@ func checkUnique(ch chan string, typ string, n int) error {
 }
 
 func TestRealtimeClient_50clients(t *testing.T) {
```
Comment (ORBAT) [inline ably/realtime_client_test.go]: Might as well rename the test

File: ably/internal/ablyutil/msgpack.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package ablyutil
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/ugorji/go/codec"
```
Comment (ORBAT) [inline ably/internal/ablyutil/msgpack.go]: Should be vendored

Comment (gernest) [reply ably/internal/ablyutil/msgpack.go]: No, not in this PR. I plan to make Godep directory go away by the time we integrate the integration branch to develop and use the new `go module` to manage dependencies.

But that work will be done later.

Comment (ORBAT) [reply ably/internal/ablyutil/msgpack.go]: Right, makes sense

**Comment by ORBAT** [comment]
As a note, this PR was necessary because the original msgpack decoder would only decode double values to a `float64`, so e.g. `1` would fail to decode

**Pull request merged**